src/user/user.controller.ts

```diff
@@ -1,33 +1,56 @@
-import { Controller, Get, Param, Post, Put, Body } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Param,
+  Post,
+  Put,
+  Body,
+  UseGuards,
+  Request,
+  Version,
+} from '@nestjs/common';
 import { UserService } from './user.service';
 import { User } from './entities/user.entity';
 import { CreateUserDto } from './dto/create-user.dto';
+import { JwtAuthGuard, RequestWithUser } from '../auth/guards/jwt-auth.guard';
 import { UpdateUserDto } from './dto/update-user.dto';
+import { API } from '../constants';
 
 @Controller('user')
 export class UserController {
   constructor(private readonly userService: UserService) {}
 
+  @Version(API.VERSION)
```

Comment:
set version global để apply cho toàn bộ controller
nếu controller nào sau này có thay đổi version thì mới vào để custom riêng cho controller/request đó